src/std-types/word_counter.md

```diff
@@ -0,0 +1,93 @@
+---
+minutes: 20
+---
+
+# Exercise: Word Counter
+
+Create a program that counts the frequency of words in a given text. The program should:
+
+1. Take a string of text as input
+2. Split the text into words (consider words to be separated by whitespace)
+3. Count how many times each word appears (case-insensitive)
+4. Print the words and their counts in alphabetical order
+
+Use a `HashMap` to store the word counts.
+
+## Example
+
+```rust
+fn main() {
+    let text = "the quick brown fox jumps over the lazy dog";
+    let counts = count_words(text);
+    print_word_counts(&counts);
+}
+```
+
+Expected output:
+```
+brown: 1
+dog: 1
+fox: 1
+jumps: 1
+lazy: 1
+over: 1
+quick: 1
+the: 2
+```
+
+## Tasks
+
+1. Implement the `count_words` function that takes a string slice and returns a `HashMap<String, u32>`
+2. Make the word counting case-insensitive (e.g., "The" and "the" count as the same word)
+3. Implement the `print_word_counts` function that prints the word counts in alphabetical order
+4. Add tests for:
```

Comment:
We haven't talked about testing yet. Most exercises focus on getting the task solved, rather than testing.